rewrite-core/src/main/java/org/openrewrite/Checksum.java

```diff
@@ -99,14 +100,20 @@ public static SourceFile checksum(SourceFile sourceFile, @Nullable String algori
 
         try {
             MessageDigest md = MessageDigest.getInstance(algorithm);
-            try (InputStream is = Files.newInputStream(sourceFile.getSourcePath());
-                 DigestInputStream dis = new DigestInputStream(is, md)) {
+            InputStream is;
+            if (sourceFile instanceof Remote) {
+                is = ((Remote) sourceFile).getInputStream(new InMemoryExecutionContext());
```

Comment:
I had thought about that as well. The `RemoteArtifactCache` is in the context and could have a different location, so that could be beneficial to reuse the existing context down here at some point.